dlp/snippets/deid/deid_table_fpe.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package deid
+
+// [START dlp_deidentify_table_fpe]
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyTableFPE de-identifies table data with format preserving encryption.
+func deidentifyTableFPE(w io.Writer, projectID string, kmsKeyName, wrappedAesKey string) error {
+	// projectId := "your-project-id"
+	/* kmsKeyName := /* keyFileName :=  "projects/YOUR_PROJECT/"
+	   + "locations/YOUR_KEYRING_REGION/"
+	   + "keyRings/YOUR_KEYRING_NAME/"
+	   + "cryptoKeys/YOUR_KEY_NAME"
+	*/
+	// wrappedAesKey := "YOUR_ENCRYPTED_AES_256_KEY"
+	// table := "your-table"
+
+	// define your table.
+	row1 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "11111"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "2015"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "$10"}},
+		},
+	}
+
+	row2 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
```

Comment:
Please use unique values for IDs.

Reply:
- I have updated the values of IDs @annadata, but just letting you know that I have used values from https://cloud.google.com/dlp/docs/samples/dlp-deidentify-table-fpe#dlp_deidentify_table_fpe-java this existing sample, that's why I have used these values.